### ISSUE TITLE
/user/<username>/history needs browser history memory

### ISSUE THREAD

**Issue by mvl22** [openstreetmap/openstreetmap-website#1089]
The /user/<username>/history page is painful when trying to find a changeset made a while back.

Ideally it would have an auto-scroll (like Facebook) when you reach the end of the current pagination, rather than requiring clicking a button.

But the main issue is that clicking on an entry, then clicking back, loses the current pagination position, so requires lots of re-clicking to get back to the same position.

**Comment by polarbearing** [comment]
Auto-scroll is not ideal either, trying to find and refer to a sequence of changeset in midst of thousands is impossible. We should go back to classic pagination, and start with page one on the oldest changesets, as it is typical in forums.

**Comment by gileri** [comment]
@polarbearing I think that the latest changesets of a user are the most relevant when looking at a user's history. Older changeset have more chance to have been overwritten by the same contributor or another user.

**Comment by polarbearing** [comment]
How you number the pages is independent from the relevance. You can start showing the page with the highest number first, of course. Just the page number remains predictable in my proposal.

**Comment by gileri** [comment]
You're right, nothing forbid to display the last (most recent) page by default, sorry for the noise.

**Comment by polarbearing** [comment]
> @tomhughes said in #933: it wouldn't get merged, because it requires every view to count all the matching changesets.

where is the difficulty? The user page ```www.openstreetmap.org/user/<name>``` gives the number of edits already. On a user history page, the most recent page number, in numbered pagination, is =round_up(edits/page_length). 

With the current endless paging, it is practically impossible to find older pages of a power user, since I cannot make a manual binary search in his pages. I can only press more and more and more until my browser explodes.

There was numbered pagination a while ago, why has it been given up?

**Comment by tomhughes** [comment]
That value on the user page is actually a special cache, not the real count.

Numbered pagination needs to go pretty much everywhere, precisely because it requires all the records to be counted to determine how many pages there are and then when page N is visited you have to read through all the records in order to find the right ones. That's why it gets noticeably slower the further down the list you go, and it's far worse when a search engine decides to visit all the pages...

The correct solution is to do pagination based on an indexed field (either the changeset ID or the timestamp in this case) so that viewing the hundredth page can be done by an indexed lookup. That is in fact exactly how this particular already works I believe, it just doesn't put that marker value in the URL at the moment.
